javalib/src/main/scala/java/net/URI.scala

```diff
@@ -1,33 +1,1415 @@
 package java.net
 
-final class URI(scheme: String,
-                userInfo: String,
-                host: String,
-                port: Int,
-                path: String,
-                query: String,
-                fragment: String)
-    extends Serializable {
+import java.io.IOException
+import java.io.Serializable
+import java.io.UnsupportedEncodingException
+import java.util.StringTokenizer
 
-  def getFragment(): String =
-    fragment
+object URI {
+  val unreserved: String = "_-!.~\'()*" //$NON-NLS-1$
 
-  def getHost(): String =
-    host
+  val punct: String = ",;:$&+=" //$NON-NLS-1$
 
-  def getPath(): String =
-    path
+  val reserved: String = punct + "?/[]@" //$NON-NLS-1$
 
-  def getPort(): Int =
-    port
+  val someLegal: String = unreserved + punct
 
-  def getQuery(): String =
-    query
+  val queryLegal: String = unreserved + reserved + "\\\"" //$NON-NLS-1$
 
-  def getScheme(): String =
-    scheme
+  val allLegal: String = unreserved + reserved
 
-  def getUserInfo(): String =
-    userInfo
+  /**
+   * Parses the given argument {@code uri} and creates an appropriate URI
+   * instance.
+   *
+   * @param uri
+   *            the string which has to be parsed to create the URI instance.
+   * @return the created instance representing the given URI.
+   */
+  def create(uri: String): URI = {
+    var result: URI = null
+    result = new URI(uri)
+    result
+  }
+
+}
+
+final class URI extends Comparable[URI] with Serializable {
+
+  import URI._
+
+  private val serialVersionUID = -6052424284110960213l
+
+  private var string: String = _
+
+  @transient private var scheme: String             = _
+  @transient private var schemespecificpart: String = _
+  @transient private var authority: String          = _
+  @transient private var userinfo: String           = _
+  @transient private var host: String               = _
+  @transient private var port: Int                  = -1
+  @transient private var path: String               = _
+  @transient private var query: String              = _
+  @transient private var fragment: String           = _
+  @transient private var opaque: Boolean            = _
+  @transient private var absolute: Boolean          = _
+  @transient private var serverAuthority: Boolean   = false
+  @transient private var hash: Int                  = -1
+
+  /**
+   * Creates a new URI instance according to the given string {@code uri}.
+   *
+   * @param uri
+   * the textual URI representation to be parsed into a URI object.
+   * @throws URISyntaxException
+   * if the given string { @code uri} doesn't fit to the
+   *                             specification RFC2396 or could not be parsed correctly.
+   */
```

Comment:
It looks like there's an implementation missing here. Did you copy/paste the javadoc twice or did you forget a method?